pkg/apis/policy/v1alpha1/types_certificaterequestpolicy.go

```diff
@@ -172,118 +174,123 @@ type CertificateRequestPolicyAllowedX509Subject struct {
 	// +optional
 	PostalCodes *CertificateRequestPolicyAllowedStringSlice `json:"postalCodes,omitempty"`
 
-	// SerialNumber defines the X.509 Subject Serial Number that may be requested
-	// for.
+	// SerialNumber defines the X.509 Subject Serial Number that may be
+	// requested for.
 	// +optional
 	SerialNumber *CertificateRequestPolicyAllowedString `json:"serialNumber,omitempty"`
 }
 
-// CertificateRequestPolicyAllowedStringSlice represents an allowed string
-// slice value paired with whether the field is a required value on the
-// request.
+// CertificateRequestPolicyAllowedStringSlice represents allowed string values
+// paired with whether the field is a required value on the request.
+// An empty Values paired with Required `true` is an "always-false" condition.
+// always denies.
+// The validating webhook enforces that Values cannot be omitted if Required
+// is `true`.
 type CertificateRequestPolicyAllowedStringSlice struct {
-	// Defines the values that are permissible to be present on request.
+	// Values defines allowed attribute values on the related
+	// CertificateRequest field.
 	// Accepts wildcards "*".
-	// An omitted field or value of `nil` forbids any value on the related field
-	// in the request from being requested.
-	// An empty slice `[]` is equivalent to `nil`, however an empty slice paired
-	// with Required `true` is an impossible condition that always denies.
-	// Values may not be `nil` if Required is `true`.
+	// If set, the related field must contain a subset of the allowed values.
+	// If `[]` or unset, the related field must be empty.
+	//
+	// NOTE: If `Required` is `true`, this field cannot be omitted. Setting the
+	// field to `[]` instead, results in an always-false expression.
```

Comment:
Not sure what `always-false` means here -- does it mean this policy will always reject?

Reply:
Yes, the wording here was originally suggested by @inteon. In the current docs, the term "impossible condition" is used. I think this is a well-known term in the policy engine domain. Maybe we should stick to that? WDYT @jsoref? Is it better or worse?